### PR TITLE
Influxdb incorrect storage types

### PR DESCRIPTION
**Pull request by geordish**
Resubmit of PR #3320

Version 2 of sending the correct data to influx. Also catching the exception that would stop updates working.

@LaurentDumont do you want to give this fix a go?

### CODE REVIEW

**Comment by LaurentDumont** [comment]
Ouf, sorry about the delay, I was running the Lan ETS in Montreal and I'm not a zombie ;) - We did end up using LibreNMS and your branch for the monitoring but I couldn't find the time to pull the latest PR. I can definitely test the package this week end poke back.

File: includes/polling/netstats-snmp.inc.php [hunk 1]
```diff
@@ -54,7 +54,7 @@
             $value = $data_array[0][$oid];
         }
         else {
-            $value = 'U';
+            $value = null;
```
Comment (laf) [inline includes/polling/netstats-snmp.inc.php]: Any reason this is changed?

File: includes/functions.php [hunk 2]
```diff
@@ -1252,10 +1252,28 @@ function function_check($function) {
...
+    * I've left the detection statement in there for a possible change in future,
+    * but currently everything just gets set to a float.
+    */
+
+    if (is_numeric($data)) {
+        // If it is an Integer
+        if (ctype_digit($data)) {
+            return floatval($data);
+        // Else it is a float
+        } else {
```
Comment (laf) [inline includes/functions.php]: Can you change any } else {

to

}
else {

**Comment by laf** [comment]
Couple of comments and then we can merge

File: includes/polling/netstats-snmp.inc.php (hunk 1 above)
Comment (geordish) [inline includes/polling/netstats-snmp.inc.php]: TBH, this entire loop can probably go. Its duplicating what is already done in includes/rrdtool.inc.php:rrdtool_update

This was changed as this will create the influx field as a string if it is the first data to be inserted. This is no good when the actual data if it is not defined is a numeric value.

File: includes/functions.php [hunk 3]
```diff
@@ -1252,10 +1252,29 @@ function function_check($function) {
...
+    if (is_numeric($data)) {
+        // If it is an Integer
+        if (ctype_digit($data)) {
+            return floatval($data);
+        // Else it is a float
+        }
+        else {
+            return floatval($data);
+        }
+    } else {
```
Comment (laf) [inline includes/functions.php]: Another } else { :)

File: includes/polling/netstats-snmp.inc.php (hunk 1 above)
Comment (laf) [inline includes/polling/netstats-snmp.inc.php]: The U is for rrdtools' benefit rather than influx. Think it needs to go back to U.

**Comment by laf** [comment]
The coding standards makes it nicer when adding more } elseif { statements as less code is changed to accomplish this leading to cleaner and clearer diff. You'll get over it.

File: includes/polling/netstats-snmp.inc.php (hunk 1 above)
Comment (geordish) [inline includes/polling/netstats-snmp.inc.php]: Sorry, I wasn't too clear.

In the rrd_update function which is called a little later, the following occurs:

```php
        foreach ($options as $k => $v) {
            if (!is_numeric($v)) {
                $v = U;
            }

            $values[] = $v;
        }
```

This sets U for non numeric data before inserting into the RRD. This bit of code here is doing that again, but on a specific subset of the data.

In my opinion, munging the data for inclusion into a datastore should belong in that datastores insert function.

Comment (laf) [inline includes/polling/netstats-snmp.inc.php]: In that case either revert the change and leave it as it was or remove the code if it's not needed.
